server/plugins.go

```diff
@@ -0,0 +1,168 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	mattermostModel "github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/matterwick/model"
+)
+
+func (s *Server) uploadPluginIfNecessary(mmClient *mattermostModel.Client4, pr *model.PullRequest) error {
+	if !strings.HasPrefix(pr.RepoName, "mattermost-plugin-") || pr.RepoName == "mattermost-plugin-api" {
+		return nil
+	}
+
+	mlog.Debug("Preparing to upload plugin to the SpinWick server")
+	mlog.Debug("Getting CI check information")
+
+	gh := github.NewClient(nil)
+	checkName := "ci"
+	checks, _, err := gh.Checks.ListCheckRunsForRef(context.Background(), pr.RepoOwner, pr.RepoName, pr.Sha, &github.ListCheckRunsOptions{CheckName: &checkName})
+	if err != nil {
+		return err
+	}
+
+	if len(checks.CheckRuns) == 0 {
+		return errors.New("checks.CheckRuns has len 0")
+	}
+
+	run := *checks.CheckRuns[0].ExternalID
+	workflowData := map[string]string{}
+	err = json.Unmarshal([]byte(run), &workflowData)
+	if err != nil {
+		return err
+	}
+
+	mlog.Debug("Getting artifacts URL")
+
+	workflowID := workflowData["workflow-id"]
+
+	jobName := "plugin-ci/build"
+	if pr.RepoName == "mattermost-plugin-playbooks" {
```

Comment:
See this section @hanzei which must be maintained per plugin or standardized.

Reply:
@DSchalla @hanzei

The plugin CI setup is identical across plugin projects (including any new plugin created from the starter template), besides the playbooks plugin project, which is why this check is here. As long as we want to use this feature before migrating to GitHub actions, this is the only special-case block we need in this regard.

2/5 When we migrate to use GitHub actions, then we can synchronize on the differences in CI and remove this special-case check.